pkg/porter/templates/schema.json

```diff
@@ -1,15 +1,134 @@
+          "type": "string"
+        },
+        "required": {
+          "default": false,
+          "description": "Indicates whether this credential must be supplied. By default, credentials are optional.",
+          "type": "boolean"
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "type": "object"
+    },
+    "output": {
+      "description": "A value that is produced by running an invocation image",
+      "properties": {
+        "applyTo": {
+          "description": "An optional exhaustive list of actions producing this output",
+          "items": {
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "definition": {
+          "description": "The name of a definition that describes the schema structure of this output",
+          "type": "string"
+        },
+        "description": {
+          "description": "A user-friendly description of this output",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of this output",
+          "type": "string"
+        },
+        "sensitive": {
+          "default": false,
+          "description": "Indicates where this output's value is sensitive and should not be logged.",
+          "type": "boolean"
+        }
+      },
+      "required": [
+        "definition",
+        "name"
+      ],
+      "type": "object"
+    },
+    "parameter": {
+      "description": "A parameter that can be passed into the invocation image",
+      "properties": {
+        "definition": {
+          "description": "The name of a definition that describes the schema structure of this parameter",
+          "type": "string"
+        },
+        "description": {
+          "description": "A user-friendly description of this output",
+          "type": "string"
+        },
+        "destination": {
+          "properties": {
+            "env": {
+              "description": "The environment variable name, such as MY_VALUE, in which the parameter value is stored",
+              "type": "string"
+            },
+            "path": {
+              "description": "The path inside of the invocation image where parameter data is mounted",
+              "type": "string"
+            }
+          },
+          "type": "object"
+        },
+        "name": {
+          "description": "The name of this parameter",
+          "type": "string"
+        },
+        "sensitive": {
+          "default": false,
+          "description": "Indicates where this parameter's value is sensitive and should not be logged.",
+          "type": "boolean"
+        }
+      },
+      "required": [
+        "definition",
+        "name"
+      ],
+      "type": "object"
+    }
+  },
   "properties": {
-    "name": {
-      "type": "string"
+    ".*": {
+      "type": "array",
+      "items": {
+        "anyOf": []
+      }
+    },
+    "credentials": {
+      "additionalProperties": {
```

Comment:
Would you switch to the pattern that we have been using to reference json schema definitions? I'd like to be consistent instead of introducing that other way the cnab-spec went with.

```json
"credentials": { "$ref": "#/definitions/credential" },
```

This applies to `credentials`, `outputs`, and the other new entries added to the top level properties in schema.json